packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterForm.tsx

```diff
@@ -421,7 +458,7 @@ const GridFilterForm = React.forwardRef<HTMLDivElement, GridFilterFormProps>(
             inputProps={{
               'aria-label': apiRef.current.getLocaleText('filterPanelLogicOperator'),
             }}
-            value={multiFilterOperator}
+            value={multiFilterOperator ?? ''}
```

Comment:
`Select` logs "switching from controlled to uncontrolled" error if `undefined` is passed as a value here